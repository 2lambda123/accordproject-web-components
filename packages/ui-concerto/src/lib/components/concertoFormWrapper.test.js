/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import ConcertoForm from './concertoForm';

const fs = require('fs');

const readOnly = false;
const type = 'test.Person';
const options = {
  includeOptionalFields: true,
  includeSampleData: 'sample',
  updateExternalModels: true,
  hiddenFields: [
    'org.accordproject.base.Transaction.transactionId',
    'org.accordproject.cicero.contract.AccordContract.contractId',
    'org.accordproject.cicero.contract.AccordClause.clauseId',
    'org.accordproject.cicero.contract.AccordContractState.stateId',
  ],
};
/**
 * Get the name and contents of all model test files
 * @returns {*} an array of name/contents tuples
 */
function getModelFiles() {
  const result = [];
  const files = fs.readdirSync(`${__dirname}/../../../test/data`);
  files.forEach((file) => {
    if (file.endsWith('.cto')) {
      const contents = fs.readFileSync(`${__dirname}/../../../test/data/${file}`, 'utf8');
      const json = fs.readFileSync(`${__dirname}/../../../test/data/${file.replace('cto', 'json')}`, 'utf8');
      result.push([file, contents, json]);
    }
  });
  return result;
}
describe('render form', () => {
  getModelFiles().forEach(([file, modelText, json]) => {
    it(`creates a React form for model ${file}`, async () => {
      const { container } = render(
        <ConcertoForm
          readOnly={readOnly}
          models={[modelText]}
          options = {options}
          type={type}
          json={JSON.parse(json)}
          onValueChange={(json) => json}
      />
      );
      await waitFor(() => screen.getByText('Name'));
      expect(container).toMatchSnapshot();
    });
  });
});
