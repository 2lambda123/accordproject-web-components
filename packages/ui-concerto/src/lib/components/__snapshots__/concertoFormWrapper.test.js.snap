// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render form, default 1`] = `"<form style=\\"min-height: 100px;\\" class=\\"ui form\\"><div class=\\"required field\\"><label>S</label><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"text\\" value=\\"Ullamco eiusmod laborum.\\"></div></div><div class=\\"field\\"><label>B</label><div class=\\"ui checked fitted read-only toggle checkbox\\"><input class=\\"hidden\\" readonly=\\"\\" tabindex=\\"0\\" type=\\"checkbox\\" value=\\"\\" checked=\\"\\"><label></label></div></div><div class=\\"required field\\"><label>Dt</label><div class=\\"field\\"><div class=\\"ui icon input\\"><i aria-hidden=\\"true\\" class=\\"calendar icon\\"></i><input readonly=\\"\\" type=\\"text\\" tabindex=\\"0\\" value=\\"2019-04-03T10:26:11.168+01:00\\"></div></div></div><div class=\\"required field\\"><label>I</label><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"number\\" value=\\"27587\\"></div></div><div class=\\"required field\\"><label>D</label><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"number\\" value=\\"70.808\\"></div></div><div class=\\"required field\\"><label>Bar</label><div class=\\"required field\\"><label>T</label><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"text\\" value=\\"Enim.\\"></div></div><div class=\\"required field\\"><label>S</label><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"text\\" value=\\"Magna amet sit.\\"></div></div></div><div class=\\"required field\\"><label>Ss</label><div class=\\"arrayElement grid\\"><div><div class=\\"required field\\"><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"text\\" value=\\"Consectetur enim laborum Lorem fugiat.\\"></div></div></div><div><button class=\\"ui basic icon negative disabled button\\" disabled=\\"\\" tabindex=\\"-1\\"><i aria-hidden=\\"true\\" class=\\"times icon\\"></i></button></div></div><div class=\\"arrayElement grid\\"><div></div><div><button class=\\"ui basic icon positive disabled button\\" disabled=\\"\\" tabindex=\\"-1\\"><i aria-hidden=\\"true\\" class=\\"plus icon\\"></i></button></div></div></div><div class=\\"required field\\"><label>Bs</label><div class=\\"arrayElement grid\\"><div><div class=\\"required field\\"><div class=\\"ui fitted read-only toggle checkbox\\"><input class=\\"hidden\\" readonly=\\"\\" tabindex=\\"0\\" type=\\"checkbox\\" value=\\"\\"><label></label></div></div></div><div><button class=\\"ui basic icon negative disabled button\\" disabled=\\"\\" tabindex=\\"-1\\"><i aria-hidden=\\"true\\" class=\\"times icon\\"></i></button></div></div><div class=\\"arrayElement grid\\"><div></div><div><button class=\\"ui basic icon positive disabled button\\" disabled=\\"\\" tabindex=\\"-1\\"><i aria-hidden=\\"true\\" class=\\"plus icon\\"></i></button></div></div></div><div class=\\"required field\\"><label>Dts</label><div class=\\"arrayElement grid\\"><div><div class=\\"required field\\"><div class=\\"field\\"><div class=\\"ui icon input\\"><i aria-hidden=\\"true\\" class=\\"calendar icon\\"></i><input readonly=\\"\\" type=\\"text\\" tabindex=\\"0\\" value=\\"2019-04-03T10:26:11.168+01:00\\"></div></div></div></div><div><button class=\\"ui basic icon negative disabled button\\" disabled=\\"\\" tabindex=\\"-1\\"><i aria-hidden=\\"true\\" class=\\"times icon\\"></i></button></div></div><div class=\\"arrayElement grid\\"><div></div><div><button class=\\"ui basic icon positive disabled button\\" disabled=\\"\\" tabindex=\\"-1\\"><i aria-hidden=\\"true\\" class=\\"plus icon\\"></i></button></div></div></div><div class=\\"required field\\"><label>Bars</label><div class=\\"arrayElement grid\\"><div><div class=\\"required field\\"><div class=\\"required field\\"><label>T</label><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"text\\" value=\\"Laborum.\\"></div></div><div class=\\"required field\\"><label>S</label><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"text\\" value=\\"Ut mollit.\\"></div></div></div></div><div><button class=\\"ui basic icon negative disabled button\\" disabled=\\"\\" tabindex=\\"-1\\"><i aria-hidden=\\"true\\" class=\\"times icon\\"></i></button></div></div><div class=\\"arrayElement grid\\"><div></div><div><button class=\\"ui basic icon positive disabled button\\" disabled=\\"\\" tabindex=\\"-1\\"><i aria-hidden=\\"true\\" class=\\"plus icon\\"></i></button></div></div></div><div class=\\"required field\\"><label>N</label><div class=\\"required field\\"><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"text\\" value=\\"ONE\\"></div></div></div><div class=\\"required field\\"><label>Ref</label><div class=\\"ui input\\"><input readonly=\\"\\" type=\\"text\\" value=\\"resource:org.hyperledger.concerto.form.test.MyAsset#2256\\"></div></div></form>"`;
